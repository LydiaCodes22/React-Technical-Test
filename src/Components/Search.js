import React, { useState } from "react";
import "../Styles/search.css";

const Search = ({getImages, setImages}) => {
      const [formData, setFormData] = useState("");
      const handleSubmit = (event) => {
        event.preventDefault();
        getImages(
          `https://images-api.nasa.gov/search?q=${formData}`, setImages
        )
        setFormData("")
      };
      const handleChange = (event) => {event.preventDefault(); setFormData(event.target.value)};

  return (
    <>
    <div className="search">
        <form onSubmit={handleSubmit}>
            <input type="text" value={formData} onChange={handleChange} placeholder="Search"/>
          <input type="submit" value="Submit" />
        </form>
      </div></>
  );
}

export default Search;