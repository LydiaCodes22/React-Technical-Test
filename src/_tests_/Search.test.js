import { render } from '@testing-library/react';
import Search from '../Components/Search';

describe('the search component', () => {
    it("renders correctly", () => {
        const { asFragment } = render(
          <Search />
        );
    
        expect(asFragment()).toMatchSnapshot();
      });
})