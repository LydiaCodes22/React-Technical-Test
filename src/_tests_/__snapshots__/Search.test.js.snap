// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the search component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <form>
      <input
        placeholder="Search"
        type="text"
        value=""
      />
      <input
        type="submit"
        value="Submit"
      />
    </form>
  </div>
</DocumentFragment>
`;
